components/script/dom/window.rs

```diff
@@ -541,14 +534,12 @@ impl WindowMethods for Window {
             stdout.flush().unwrap();
             stderr.flush().unwrap();
         }
-
         let (sender, receiver) = ProfiledIpc::channel(self.global().time_profiler_chan().clone()).unwrap();
-        self.send_to_constellation(ScriptMsg::Alert(s.to_string(), sender));
-
-        let should_display_alert_dialog = receiver.recv().unwrap();
-        if should_display_alert_dialog {
-            display_alert_dialog(&s);
-        }
+        let window_proxy = self.window_proxy.get().unwrap();
+        let top_level_browsing_context_id = window_proxy.top_level_browsing_context_id();
```

Comment:
We do that a lot. Like mentioned earlier, let's automatize that by automatically adding the browser context id to the message if possible.

Reply:
This occured to me, and then I realized that in order to 'instantiate' the message we need this `top_level_browsing_context_id` before we pass it to a general function like `send_to_embedder`. Is there are way to do something like `let msg = EmbedderMsg::Alert(<placeholder_to_be_set_later>, s.to_string(), sender); ?